packages/core/src/utils/__tests__/security.spec.ts

```diff
@@ -329,4 +329,65 @@ describe('validateSecurity', () => {
       });
     });
   });
+
+  describe('AND relation between security schemes', () => {
+    describe('when given only one key', () => {
```

Comment:
I see you're trying to use describe for "context" and it's good that you're trying, but this seems a bit off. Generally the context is to provide meaningful setup for multiple tests. The action is not in the context, so "when given one key" is kinda in the wrong place.

I would do it more like this:

```
  describe('AND relation between security schemes', () => {
    describe('when security scheme expects two keys', () => {
      const securityScheme = [
        [
          {
            in: 'header',
            type: 'apiKey',
            name: 'x-api-key',
          },
          {
            in: 'query',
            type: 'apiKey',
            name: 'apiKey',
          },
        ],
      ];

      it('fails with an invalid security scheme error', () => {
        expect(
          validateSecurity<any, any>(
            { headers: { 'x-api-key': 'abc123' } },
            { security: securityScheme },
          ),
        ).toStrictEqual([
          {
            headers: {},
            message: 'Invalid security scheme used',
            name: 'Unauthorised',
            status: 401,
          },
        ]);
      });
    });
```

This has a few benefits. 

1. setup is seperate from assertion, so its easier to read what is going on, 
2. people can easily add similar tests/assertions using the same setup, reducing accidental false positives of badly copied and pasted code
3. the describe is describing itself :)

Can you have a go at this?